src/app.js

```diff
@@ -0,0 +1,29 @@
+go.app = function() {
+    var vumigo = require('vumigo_v02');
+    var App = vumigo.App;
+    var EndState = vumigo.states.EndState;
+    var utils = require('./utils');
+
+    var GoApp = App.extend(function(self) {
+        App.call(self, 'states:start');
+        
+        self.states.add('states:start', function(name) {
+            state = new EndState(name, {
```

Comment:
You probably want to structure this slightly differently and you don't need to override `.display`. It could look something like:

```
self.states.add('states:start', function(name) {
    var matching_keywords = utils.find_keywords(self.im.msg);
    var p = Q();
    if (!_.isEmpty(matching_keywords)) {
        p = p.then(function () {
            return send_email(self.im.config.email_address, matching_keywords));
        });
    }
    return p.thenResolve(new EndState(name, {
        ...
    });
});
```
Hopefully I wrote down all the promise machinery correctly from memory.